src/plugins/data_views/common/data_views/data_view.ts

```diff
@@ -272,15 +272,19 @@ export class DataView implements DataViewBase {
   /**
    * Creates static representation of the data view.
    */
-  public toSpec(): DataViewSpec {
-    return {
+  public toSpec(includeFields = true): DataViewSpec {
```

Comment:
I would argue to remove this argument or I suggest leaving the default to `false`.

I know this theoretically breaks the backward compatibility, but practically it does not. At the moment every time we create DataView from spec we refetch the field list (we don't reuse the one from the spec). Also nobody seems to be directly using the field list from the spec. 

Which means we are at the moment supporting a feature that is not needed. We are keeping support for that unneeded feature and also making it the default behavior. This will lead to a lot of field lists stored without a reason.

I think we should remove it and introduce it when we have a requirement for it, or at least by default not include it to make sure we only include the field list when the consumer actually needs it.

Reply:
Currently there are already places that rely on this behavior (for example, Lens). I'll have to do some digging to find out why this behavior is relied on, but setting the default to false breaks a lot of pieces in Kibana.

Reply:
ok i was not aware that we have places that expect fields on the spec already.